R/get_table.R

```diff
@@ -10,7 +10,10 @@
 #'   Default FALSE.
 #'   If set TRUE, the history columns "checksum", "from_ts", "until_ts" are returned also
 #' @return
-#'   A "lazy" dataframe (tbl_lazy) generated using dbplyr
+#'   A "lazy" dataframe (tbl_lazy) generated using dbplyr.
+#'
+#'   Note that a temporary table will be preferred over ordinary tables in the default schema (see [get_schema()]) with
```

Comment:
The order is correct and is a consequence of `dplyr` preferring temporary tables over ordinary tables:

``` r
conn <- DBI::dbConnect(RSQLite::SQLite())

data.frame(source_schema = "main") |>
  dplyr::copy_to(dest = conn,
                 df = _,
                 name = DBI::Id(schema = "main", table = "t"),
                 temporary = FALSE,
                 overwrite = TRUE)

data.frame(source_schema = "temp") |>
  dplyr::copy_to(dest = conn,
                 df = _,
                 name = DBI::Id(schema = "temp", table = "t"),
                 temporary = FALSE,
                 overwrite = TRUE)

dplyr::tbl(conn, "t")
#> # Source:   table<t> [1 x 1]
#> # Database: sqlite 3.43.2 []
#>   source_schema
#>   <chr>        
#> 1 temp
```
Now, we outrule the order of creation determining preference
```r
local({
  DBI::dbRemoveTable(conn, "t")
  DBI::dbRemoveTable(conn, "t")
})

# Ensure both tables are correctly removed
nrow(DBI::dbGetQuery(conn, "SELECT * FROM pragma_table_list WHERE name == 't'")) == 0
#> [1] TRUE

data.frame(source_schema = "temp") |>
  dplyr::copy_to(dest = conn,
                 df = _,
                 name = DBI::Id(schema = "temp", table = "t"),
                 temporary = FALSE,
                 overwrite = TRUE)

data.frame(source_schema = "main") |>
  dplyr::copy_to(dest = conn,
                 df = _,
                 name = DBI::Id(schema = "main", table = "t"),
                 temporary = FALSE,
                 overwrite = TRUE)

dplyr::tbl(conn, "t")
#> # Source:   table<t> [1 x 1]
#> # Database: sqlite 3.43.2 []
#>   source_schema
#>   <chr>        
#> 1 temp
```

<sup>Created on 2023-12-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Hmm. I guess my confusion stems from how `get_schema` is written for `SQLite` where `main` is the schema being used as the primary schema and that here it is reversed.

I think this behaviour of using the temporary tables when tie breaking is a weird choice in `dplyr/RSQLite`, but that might just be me.

However, if this the "standard" behaviour, it may best to follow this throughout (i.e. change the order in `get_schema.SQLiteConnection` and put the same disclaimer in the documentation of `get_schema`

Reply:
In order to avoid this PR hitting commit #&ZeroWidthSpace;50, this behavior will stay in order to comply with `dplyr` preference and is now within the scope of #76.